basic literacy.i7x

```diff
@@ -7,110 +7,118 @@ Basic Literacy by Bart Massey begins here.
 
-Rule for implicitly taking an erasing tool (this is the implicitly take an erasing tool rule): let E be a random erasing tool; silently try taking E; say implicitly taking E.
-
-Check erasing (this is the check erasing rule): if the noun is not writable or the read text of the noun is empty, instead say error the noun need not be erased; if the noun is not writable, instead say error the noun is not erasable; if the player carries an erasing tool, rule succeeds; if an erasing tool is touchable, abide by the implicitly take an erasing tool rule; instead say error no erasing tool.
+Before erasing when the player does not carry an erasing tool and an erasing tool is touchable (this is the obtain an erasing tool rule):
+	repeat with E running through touchable erasing tools:
+		say implicitly taking E;
+		silently try taking E;
+		if rule succeeded:
+			continue the action;
+	say error failed to obtain an erasing tool;
+	rule fails.
 
-Carry out erasing (this is the erasing rule): now the read text of the noun is "".
+Check an actor erasing (this is the check erasing rule): unless the noun is erasable, instead say "[error the noun cannot be erased]" (A); unless the read text of the noun is not empty, instead say "[error the noun need not be erased]" (B); unless the player carries an erasing tool, instead say "[error no erasing tool]" (C).
 
+Carry out an actor erasing (this is the carry out erasing rule): now the read text of the noun is "" (A).
 Report erasing (this is the report erasing rule): say "You erase [the noun]."
 
 Basic Literacy ends here.
 
 ---- Documentation ----
-
 This module extends Inform 7 by providing some basic actions and types that allow the player to read, write and erase text. This is implemented by giving every thing an indexed text called the "read text". With this module loaded, the "read" command no longer invokes the "examining" action. Instead, "read [thing]" shows the read text of the thing read.
-
 Allowing the player to write arbitrary text on things is also allowed. The thing to be written on must have the "writable" property, and the player must have access to a "writing tool". Subsequent writes will append new paragraphs to the read text of the thing written.
+Allowing the player to erase the text on things is also allowed. The thing to be erased on must have the "erasable" property, and the player must have access to an "erasing tool". A writable thing is also erasable unless stated otherwise. 
+Writable-Erasable things: A notebook, a whiteboard, a graffiti
+Writable-only things: An engraving, a legal document.
 
-Allowing the player to erase the text on things is also allowed. The thing to be written on must have the "writable" property, and the player must have access to an "erasing tool".
-
-Example: ** OLAB - A whiteboard on which the player can make notes
-
-This is a really straightforward example.
-
+Example: * OLAB - A writable erasable thing 
```

Comment:
I think my old description was better. See the Manual for other example descriptions to check what style is expected.